test/dynamics/solvers/test_solver_classes.py

```diff
@@ -1263,6 +1264,41 @@ def test_t_eval_t_span_jax_expm(self):
             max_dt=0.05,
         )
 
+    def test_jit_solve_with_internal_jit(self):
```

Comment:
Good test here. Can you make the schedule more complicated by adding `ShiftPhase`, `SetPhase`, `ShiftFrequency`, and `SetFrequency` instructions? I believe you updated the PR so that these could also be jitted through, but that is not being tested here.

Can you also add a derivative test here as well, not just `jit`.

Reply:
add the new test with https://github.com/Qiskit/qiskit-dynamics/pull/149/commits/3f214c574e2f611b635dabde4dc21e150a411697